packages/xo-web/src/common/ip-utils.js

```diff
@@ -99,24 +99,8 @@ export const formatIps = ips => {
       (splitIp1[0] - splitIp2[0]) * 256 * 256 * 256
     )
   })
-  const range = { first: '', last: '' }
-  const formattedIps = []
-  let index = 0
-  forEach(sortedIps, ip => {
-    if (ip !== getNextIpV4(range.last)) {
-      if (range.first) {
-        formattedIps[index] =
-          range.first === range.last ? range.first : { ...range }
-        index++
-      }
-      range.first = range.last = ip
-    } else {
-      range.last = ip
-    }
-  })
-  formattedIps[index] = range.first === range.last ? range.first : range
 
-  return formattedIps
+  return [{ first: sortedIps[0], last: sortedIps[sortedIps.length - 1] }]
```

Comment:
It will always display a range even in case of one IP, see https://github.com/vatesfr/xen-orchestra/blob/master/packages/xo-web/src/xo-app/settings/ips/index.js#L82

IMHO, the issue is in the function `getNextIpV4` which returns '*.1' instead of '*.0' in case of '*.255' entered IP.

Reply:
Yes. I'm not sure we should store `*.0` IPs in IP pools but at least it should be consistent: [generating the IPs from a range](https://github.com/vatesfr/xen-orchestra/blob/a186672447c14ba7d6c05e9c398410d83fe1bbcf/packages/xo-web/src/common/ip-utils.js#L32-L50) and [generating the range from the IPs](https://github.com/vatesfr/xen-orchestra/blob/a186672447c14ba7d6c05e9c398410d83fe1bbcf/packages/xo-web/src/common/ip-utils.js#L105-L117) should behave the same way.